core/src/main/java/org/wildfly/channel/Channel.java

```diff
@@ -151,4 +153,70 @@ public enum NoStreamStrategy {
         @JsonProperty("none")
         NONE
     }
+
+    /**
+     * Builder for channel class
+     */
+    public static class Builder {
```

Comment:
Is it required that this builder be `public` and part of the API?
Could we make it `package-protected so that we can use that internally (and in the tests) instead?

Reply:
I think the public builder would be useful for the Channel. The Channel has a lot of optional properties and using the public constructor is becoming a bit confusing.

But it's preferred not to expose that, I can make it package protected

Reply:
thanks for the explanation, LGTM